tst/testrandom.g

```diff
@@ -1,34 +1,38 @@
-randomTest := function(collection, method, checkin...)
-    local test1, test2, test3, test4, test5, test6, localgen, checkmethod;
-    if Length(checkin) = 0 then
-        checkmethod := \in;
-    else
-        checkmethod := checkin[1];
-    fi;
+# Perform a variety of tests on Random generators.
+# filter: The type of random source we are testing
+# globalobj: The name of the global instance of this iterator
+# gmethod: A single argument method which takes elements of the collection
+#          using globalobj (This exists to let us test single-argument Random)
+# method: Two argument random method, which takes a random source and collection
+# collection: The collection we want to look for random methods of
+# checkmethod: a way of checking if a value is a member of collection (usually \in)
+
+randomTestInner := function(filter, globalobj, gmethod, method, collection, checkmethod)
+    local test1, test2, test3, test4, test5, test6, localgen;
 
     # We do a single call first, to deal with calling Random causing extra attributes
     # of 'collection' to be set, changing the dispatch
     method(collection);
```

Comment:
Should this perhaps be `gmethod`? Otherwise, the documentation comment is wrong, as here `method` is called with a single argument.

Reply:
This is definately wrong